phosphor-auditlog/auditLogUtils.hpp

```diff
@@ -0,0 +1,105 @@
+#pragma once
+
+#include <libaudit.h>
+
+#include <phosphor-logging/lg2.hpp>
+#include <xyz/openbmc_project/Common/File/error.hpp>
+
+#include <filesystem>
+#include <fstream>
+#include <string>
+
+namespace phosphor::auditlog
+{
+
+/** @class ALParseFile
+ *  @brief Creation of temporary file for parsed audit log
+ */
+class ALParseFile
+{
+  public:
+    ALParseFile(const ALParseFile&) = delete;
+    ALParseFile& operator=(const ALParseFile&) = delete;
+    ALParseFile(ALParseFile&&) = delete;
+    ALParseFile& operator=(ALParseFile&&) = delete;
+
+    ~ALParseFile()
+    {
+        if (!pathName.empty() && !keepFile)
+        {
+            lg2::debug("Removing {FILE}", "FILE", pathName);
+            std::filesystem::remove(pathName);
+        }
+    }
+
+    /** @brief Create empty temporary file
+     */
+    ALParseFile()
+    {
+        std::string tempFile = std::filesystem::temp_directory_path() /
+                               "auditLogJson-XXXXXX";
```

Comment:
I don't think it creates a temporary directory, rather just a temp file in /tmp?

Reply:
> Also, can the default constructor ALParseFile() and ALParseFile(const std::string& filePath) be combined with the appropriate handling depending on the input filePath?

They could be combined, but they seemed enough different that it didn't seem worth it to combine them. I thought it would be much clearer to make them separate functions.